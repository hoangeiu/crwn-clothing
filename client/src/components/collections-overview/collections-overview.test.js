import React from "react";
import { shallow } from "enzyme";

import { CollectionsOverview } from "./collections-overview.component";
import CollectionPreview from "../collection-preview/collection-preview.component";

describe("CollectionsOverview component", () => {
  let wrapper;
  let mockCollections;
  beforeEach(() => {
    mockCollections = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];

    const mockProps = {
      collections: mockCollections,
    };

    wrapper = shallow(<CollectionsOverview {...mockProps} />);
  });

  it("should CollectionsOverview render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a equal number of CollectionPreview component as collections prop", () => {
    expect(wrapper.find(CollectionPreview).length).toEqual(
      mockCollections.length
    );
  });
});
