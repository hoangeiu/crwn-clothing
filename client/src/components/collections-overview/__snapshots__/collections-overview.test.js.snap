// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component should CollectionsOverview render without error 1`] = `
<div
  className="collections-overview"
>
  <withRouter(CollectionPreview)
    key="1"
  />
  <withRouter(CollectionPreview)
    key="2"
  />
  <withRouter(CollectionPreview)
    key="3"
  />
  <withRouter(CollectionPreview)
    key="4"
  />
  <withRouter(CollectionPreview)
    key="5"
  />
</div>
`;
